org.lflang/src/org/lflang/TimeValue.java

```diff
@@ -0,0 +1,168 @@
+/*************
+Copyright (c) 2019, The University of California at Berkeley.
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND 
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED 
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES 
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON 
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS 
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***************/
+
+package org.lflang;
+
+import org.jetbrains.annotations.NotNull;
```

Comment:
What is this for?

Reply:
Ah this is Intellij importing stuff automatically, I usually remove it. Basically this is an annotation you can use in Java code to make it clear that some variable is never null. There's also `@Nullable` which means that a variable may be null, and so you must always check for null before dereferencing the variable. Using those annotations improves compatibility with the kotlin type system, gives more info to static analysis tools, and helps document API, the end goal being to avoid NullPointerExceptions. But if we are to use them, we need to decide which annotations to use, as those from `org.jetbrains.annotations` are only one option.

Maybe I can add an iterm to #478?

Reply:
It may be better to use something more neutral. Doesn't Java have its own?

Reply:
No, there are `javax` ones but we should really not use those. We have [those other choices](https://kotlinlang.org/docs/java-interop.html#nullability-annotations). Those from jetbrains are available as they're part of the kotlin dependency. Maybe we can use those from the JDT instead?

Reply:
JSpecify maybe?

Reply:
Since this just works, let's keep it like this for now